### PR TITLE
ECER-996: Mark messages as READ

### PR DESCRIPTION
**Pull request by pcatkins**
## Description

- New endpoint `/api/messages/{id}/seen`
- Endpoint tests
- Repository tests
- Integrate with frontend message list view

## Related Jira Issue(s)

- [ECER-966](https://eccbc.atlassian.net/browse/ECER-996)

## Checklist
- [X] I have tested these changes locally.
- [X] Changes to backend endpoints are covered by passing integration tests.
- [ ] I have added or updated the necessary documentation.

### CODE REVIEW

File: src/ECER.Resources.Accounts/Communications/CommunicationRepository.cs [hunk 1]
```diff
@@ -42,4 +42,22 @@ public async Task<IEnumerable<Communication>> Query(UserCommunicationQuery query
...
   }
+  
+  public async Task<string> Seen(string communicationId, CancellationToken cancellationToken)
+  {
+    await Task.CompletedTask;
+
+    var communication =
+      context.ecer_CommunicationSet.Single(c => c.ecer_CommunicationId == Guid.Parse(communicationId));
+    
+    if (communication == null) throw new InvalidOperationException($"Application '{communicationId}' not found");
```
Comment (farzadnadiri) [inline src/ECER.Resources.Accounts/Communications/CommunicationRepository.cs:53]: This should be a Communication

File: src/ECER.Managers.Registry/CommunicationHandlers.cs [hunk 2]
```diff
@@ -43,4 +45,36 @@ public async Task<CommunicationsQueryResults> Handle(Contract.Communications.Use
...
+  
+  /// <summary>
+  /// Handles marking a communication as seen use case
+  /// </summary>
+  /// <param name="request">The command</param>
+  /// <param name="cancellationToken">cancellation token</param>
+  /// <returns></returns>
+  public async Task<string> Handle(CommunicationSeenCommand request, CancellationToken cancellationToken)
+  {
+    ArgumentNullException.ThrowIfNull(communicationRepository);
```
Comment (farzadnadiri) [inline src/ECER.Managers.Registry/CommunicationHandlers.cs:57]: We don't need line 57 and 58 anymore

File: src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Communications/CommunicationsEndpoints.cs [hunk 3]
```diff
@@ -25,6 +28,26 @@ public void Register(IEndpointRouteBuilder endpointRouteBuilder)
      .WithOpenApi("Handles messages queries", string.Empty, "message_get")
      .RequireAuthorization();
 
+    endpointRouteBuilder.MapPut("/api/messages/{id}/seen",
+        async Task<Results<Ok<CommunicationResponse>, BadRequest<string>>> (string? id,
+          CommunicationSeenRequest request, HttpContext ctx, IMediator messageBus, CancellationToken ct) =>
+        {
+          var userId = ctx.User.GetUserContext()?.UserId;
+
+
```
Comment (farzadnadiri) [inline src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Communications/CommunicationsEndpoints.cs:37]: Could you please remove this free space?

File: src/ECER.Tests/Integration/Resources/Accounts/Communications/CommunicationRepositoryTests.cs [hunk 4]
```diff
@@ -28,4 +31,25 @@ public async Task QueryCommunications_ById_Found()
     // Assert
     communications.ShouldHaveSingleItem();
   }
+  
+  [Fact]
+  public async Task SeenCommunications_MarkAsSeen()
```
Comment (farzadnadiri) [inline src/ECER.Tests/Integration/Resources/Accounts/Communications/CommunicationRepositoryTests.cs:36]: This probably should be MarkUnseenCommunication_AsSeen.
This test probably won't work next time as you mark the test communication as seen unless you create a new test communication every time you run this test.

File: src/ECER.Resources.Accounts/Communications/ICommunicationRepository.cs [hunk 5]
```diff
@@ -10,6 +10,7 @@ namespace ECER.Resources.Accounts.Communications;
 public interface ICommunicationRepository
 {
   Task<IEnumerable<Communication>> Query(UserCommunicationQuery query);
+  Task<string> Seen(string communicationId, CancellationToken cancellationToken);
```
Comment (farzadnadiri) [inline src/ECER.Resources.Accounts/Communications/ICommunicationRepository.cs:13]: is MarkAsSeen better?

File: src/ECER.Managers.Registry.Contract/Communications/Contract.cs [hunk 6]
```diff
@@ -2,6 +2,11 @@
 
 namespace ECER.Managers.Registry.Contract.Communications;
 
+/// <summary>
+/// Invokes communication seen use case
+/// </summary>
+public record CommunicationSeenCommand(string communicationId, string userId) : IRequest<string>;
```
Comment (farzadnadiri) [inline src/ECER.Managers.Registry.Contract/Communications/Contract.cs:8]: MarkCommunicationAsSeenCommand?

File: src/ECER.Tests/Integration/RegistryApi/CommunicationTests.cs [hunk 7]
```diff
@@ -39,6 +39,32 @@ public async Task GetMessageStatus_ReturnsStatus()
...
     var communicationsStatus = await communicationsStatusResponse.ReadAsJsonAsync<CommunicationsStatusResults>();
     communicationsStatus.ShouldNotBeNull();
-
+  }
+  
+  
+  [Fact]
+  public async Task SeenCommunication_ReturnId()
+  {
+    var communicationSeenResponse =  await Host.Scenario(_ =>
```
Comment (farzadnadiri) [inline src/ECER.Tests/Integration/RegistryApi/CommunicationTests.cs:48]: This Test will work just one time

File: src/ECER.Tests/Integration/Resources/Accounts/Communications/CommunicationRepositoryTests.cs (hunk 4 above)
Comment (SoLetsDev) [reply src/ECER.Tests/Integration/Resources/Accounts/Communications/CommunicationRepositoryTests.cs:36]: hmm the code below seems to check that the communication has not been acknowledged first. So it seems like the communication is getting "refreshed" for each test.

Comment (farzadnadiri) [reply src/ECER.Tests/Integration/Resources/Accounts/Communications/CommunicationRepositoryTests.cs:36]: Fixture.communicationId comes from RegistryPortalWebAppScenarioBase , GetOrAddCommunication and seems we don't check that there

Comment (pcatkins) [reply src/ECER.Tests/Integration/Resources/Accounts/Communications/CommunicationRepositoryTests.cs:36]: @farzadnadiri @SoLetsDev this test is passing consistently

File: src/ECER.Tests/Integration/RegistryApi/CommunicationTests.cs (hunk 7 above)
Comment (pcatkins) [reply src/ECER.Tests/Integration/RegistryApi/CommunicationTests.cs:48]: @farzadnadiri @SoLetsDev this test is **not** passing consistently

**Review by farzadnadiri — APPROVED** [review]

**Pull request merged**